docs/public/_redirects

```diff
@@ -167,6 +164,11 @@ https://v3-9-0.mui.com/* https://v3.mui.com/:splat 301!
 /components/slider-styled/ /components/slider/ 301
 /customization/styled/ /system/styled/ 301
 /api/data-grid/x-grid/ /api/data-grid/data-grid-pro/ 301
+https://next.material-ui.com/* https://mui.com/:splat 301!
+https://v0.material-ui.com/* https://v0.mui.com/:splat 301!
+https://v1.material-ui.com/* https://v1.mui.com/:splat 301!
+https://v3.material-ui.com/* https://v3.mui.com/:splat 301!
+https://v4.material-ui.com/* https://v4.mui.com/:splat 301!
```

Comment:
what's happening here?

Reply:
I'm moving the block from one header which is about supporting multiple domains (ongoing), to another which is about avoiding 404 for legacy URLs.

Reply:
Gotcha, thanks!